src/time.rs

```diff
@@ -0,0 +1,242 @@
+use crate::sys::time::TimeSpec;
+#[cfg(target_os = "linux")]
+use crate::unistd::Pid;
+#[cfg(target_os = "linux")]
+use crate::Error;
+use crate::{Errno, Result};
+use libc::{self, clockid_t};
+use std::mem::MaybeUninit;
+
+/// Clock identifier
+///
+/// Newtype pattern around `clockid_t` (which is just alias). It pervents bugs caused by
+/// accidentally passing wrong value.
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash)]
+pub struct ClockId(clockid_t);
+
+impl ClockId {
+    /// Creates `ClockId` from raw `clockid_t`
+    pub fn from_raw(clk_id: clockid_t) -> Self {
+        ClockId(clk_id)
+    }
+
+    /// Returns `ClockId` of a `pid` CPU-time clock
+    #[cfg(target_os = "linux")]
+    pub fn pid_cpu_clock_id(pid: Pid) -> Result<Self> {
+        clock_getcpuclockid(pid)
+    }
+
+    /// Returns resolution of the clock id
+    pub fn res(self) -> Result<TimeSpec> {
+        clock_getres(self)
+    }
+
+    /// Returns time on the clock id
+    pub fn time(self) -> Result<TimeSpec> {
```

Comment:
Since this method is basically the same as `std::time::Instant::now`, it would be good if the API could be similar.  Do you agree?  The `time` method could be renamed `now`, and we could add convertability between `TimeSpec` and `Instant`.

Reply:
Agreed. Wasn't sure if `now()` makes sense for pid-based clocks, but I guess it sort of does as we get the time on the pid-based clock at the current moment. I'll update the PR.

Reply:
Looking more, it seems like `std::time::Instant` is pretty limited in its usage and seems to be designed to just get the current monotonically increasing time, rather than all purpose type to represent an arbitrary point in time (i.e. the only way to construct `std::time::Instant` is to call `Instant::now()`).

The other problem is that I don't think it makes sense to convert the time process has been running on CPU to a particular point in time since unix epoch. Looking on C `timespec` [documentation](https://en.cppreference.com/w/c/chrono/timespec) I think `TimeSpec` is closer to `std::time::Duration` rather than `std::time::Instant` (or even `std::time::SystemTime`), as it represents the interval rather than particular point in time.

If that makes sense, I can add convertability between `TimeSpec` and `Duration`.

Reply:
Hm, good point.  I agree.  Duration is more appropriate.